components/camera/videosource/webcam.go

```diff
@@ -328,7 +329,7 @@ func NewWebcamSource(ctx context.Context, attrs *WebcamAttrs, logger golog.Logge
 		}
 	}, camWg.activeBackgroundWorkers.Done)
 
-	return &camWg, nil
+	return cam.(camera.Camera), nil
```

Comment:
The issue the `CameraWaitGroup` was designed to solve is this: how do we ensure the goroutine is closed before the program terminates? Or more generally, how do we avoid a goroutine leak?

This function returns a `camera.Camera` object to its caller, some concrete type that satisfies the `camera.Camera` interface. The caller will correctly call `Camera.Close()` to ensure that they've leaked no resources. The concrete type `CameraWaitGroup.Close` will wait until the above goroutine has been terminated _before_ closing the `camera.Camera` object it contains. The `camera.Camera` you return now will not wait and will immediately call its concrete type's `Close()` function without any consideration given to the above goroutine. This will cause a goroutine leak.

Reply:
I don't have a lot of context into the exact problem you're trying to solve so it's hard for me to give more helpful/specific advice 😅. If you need help brainstorming I can do my best to help if you link the ticket you're working or add more info here. If you have it figured out though that's fine too.